zerver/actions/message_send.py

```diff
@@ -887,12 +887,14 @@ class UserData(TypedDict):
             users.append(user_data)
 
         sender = send_request.message.sender
+        sender_is_welcome_bot = sender.delivery_email == settings.WELCOME_BOT
         message_type = wide_message_dict["type"]
         user_notifications_data_list = [
             UserMessageNotificationsData.from_user_id_sets(
                 user_id=user_id,
                 flags=user_flags.get(user_id, []),
                 private_message=(message_type == "private"),
+                sender_is_welcome_bot=sender_is_welcome_bot,
```

Comment:
Why did you put this here in the sequence of parameters? I think at the end might be better for readability.

Reply:
> Why did you put this here in the sequence of parameters?

* To keep boolean fields (private_message, sender_is_welcome_bot) at one place.
* Set[int] (*_user_ids) fields at one place.

Hence, Keeping it consistent with [this](https://github.com/zulip/zulip/pull/23225/files#diff-8319130d25323802a8a47fa49f43d3c5e4c73567aa1e2fc95adb6d4afa68646aL24)

Reply:
OK, seems reasonable I guess.